### ISSUE TITLE
False positive E1130 invalid-unary-operand-type from Numpy einsum

### ISSUE THREAD

**Issue by robertsont** [PyCQA/pylint#5821]
### Bug description

It seems this bug has appeared before, for example [here](https://github.com/PyCQA/pylint/issues/2436) but was fixed by [#3974](https://github.com/PyCQA/astroid/pull/918). Generally, return values from `numpy` functions are handled correctly but pylint seems to think that the return value from [`numpy.einsum`](https://numpy.org/doc/stable/reference/generated/numpy.einsum.html) is of `NoneType` rather than `ndarray`. Below is a code snippet to reproduce the error.

```python
# unary_enisum.py
# pylint: disable=missing-docstring

import numpy as np


a = np.arange(25).reshape(5, 5)
b = np.einsum("ii", a)

c = -b
```

Apolgies if I've missed an already opened issue for this and thanks for maintaining a great package!

### Command used

```shell
pylint unary_einsum.py
```

### Pylint output

```shell
************* Module unary_einsum
unary_einsum.py:9:4: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)

------------------------------------
Your code has been rated at -2.50/10
```

### Expected behavior

Passing output, i.e.

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.1 (default, Jan  1 2021, 01:07:55)
[GCC 7.5.0]
```

### OS / Environment

Ubuntu 18.04.6 LTS (Bionic Beaver)

### Additional dependencies

`numpy` version 1.22.2

**Comment by DanielNoord** [comment]
Will be fixed with `astroid` `2.12` thanks to @mbyrnepr2 😄

**Comment by DanielNoord** [comment]
This has been fixed some time ago in `astroid`. I'm closing 😄
